lib/AST/ASTContext.cpp

```diff
@@ -3302,6 +3302,13 @@ PackType *PackType::getEmpty(const ASTContext &C) {
   return cast<PackType>(CanType(C.TheEmptyPackType));
 }
 
+PackType *PackType::getSingletonPackExpansion(Type param) {
+  assert((param->is<GenericTypeParamType>() &&
```

Comment:
I think you want TypeBase::isParameterPack(), since I'm assuming you want to wrap a dependent member type `(each T).Foo` in a singleton pack also.

Reply:
Yeah, that's probably useful